lms/djangoapps/shoppingcart/views.py

```diff
@@ -820,20 +823,89 @@ def show_receipt(request, ordernum):
 
     if order.user != request.user or order.status not in ['purchased', 'refunded']:
         raise Http404('Order not found!')
 
-    if 'application/json' in request.META.get('HTTP_ACCEPT', ""):
+    if is_json_request:
         return _show_receipt_json(order)
     else:
         return _show_receipt_html(request, order)
 
 
+def _get_external_order(request, order_number):
+    """Get the order context from the external E-Commerce Service.
+
+    Get information about an order. This function makes a request to the E-Commerce Service to see if there is
+    order information that can be used to render a receipt for the user.
+
+    Args:
+        request (Request): The request for the the receipt.
+        order_number (str) : The order number.
+
+    Returns:
+        dict: A serializable dictionary of the receipt page context based on an order returned from the E-Commerce
+            Service.
+
+    """
+    try:
+        api = EcommerceAPI()
+        order_number, order_status, order_data = api.get_order(request.user, order_number)
+        billing = order_data.get('billing_address', {})
+        country = billing.get('country', {})
+
+        # In order to get the date this order was paid, we need to check for payment sources, and associated
+        # transactions.
+        payment_date = None
+        for source in order_data.get('sources', []):
+            for transaction in source.get('transactions', []):
+                payment_date = transaction['date_created']
+        order_info = {
+            'orderNum': order_number,
+            'currency': order_data['currency'],
+            'status': order_status,
+            'purchase_datetime': payment_date,
+            'total_cost': order_data['total_excl_tax'],
+            'billed_to': {
+                'first_name': billing.get('first_name', ''),
+                'last_name': billing.get('last_name', ''),
+                'street1': billing.get('line1', ''),
+                'street2': billing.get('line2', ''),
+                'city': billing.get('line4', ''),  # 'line4' is the City, from the E-Commerce Service
+                'state': billing.get('state', ''),
+                'postal_code': billing.get('postcode', ''),
+                'country': country.get('display_name', ''),
+            },
+            'items': [
+                {
+                    'quantity': item['quantity'],
+                    'unit_cost': item['unit_price_excl_tax'],
+                    'line_cost': item['line_price_excl_tax'],
+                    'line_desc': item['description']
+                }
+                for item in order_data['lines']
+            ]
+        }
+        return JsonResponse(order_info)
+    except InvalidConfigurationError:
+        msg = u"E-Commerce API not setup. Cannot request Order [{order_number}] for User [{user_id}] ".format(
+            user_id=request.user.id, order_number=order_number
+        )
+        log.debug(msg)
+        raise Http404(msg)
```

Comment:
I'm going to change it to a JSON response with a status code of 500, I think that would be more in line with the contract of this function.